ledger/accountdb.go

```diff
@@ -1026,74 +1080,395 @@ func accountsPutTotals(tx *sql.Tx, totals ledgercore.AccountTotals, catchpointSt
 	return err
 }
 
-// accountsNewRound updates the accountbase and assetcreators tables by applying the provided deltas to the accounts / creatables.
-// The function returns a dbAccountData for the modified accounts which can be stored in the base cache.
-func accountsNewRound(tx *sql.Tx, updates compactAccountDeltas, creatables map[basics.CreatableIndex]ledgercore.ModifiedCreatable, proto config.ConsensusParams, lastUpdateRound basics.Round) (updatedAccounts []dbAccountData, err error) {
+func loadHoldings(stmt *sql.Stmt, eah ledgercore.ExtendedAssetHolding) (map[basics.AssetIndex]basics.AssetHolding, ledgercore.ExtendedAssetHolding, error) {
+	if len(eah.Groups) == 0 {
+		return nil, ledgercore.ExtendedAssetHolding{}, nil
+	}
+	var err error
+	holdings := make(map[basics.AssetIndex]basics.AssetHolding, eah.Count)
+	for gi, g := range eah.Groups {
+		holdings, eah.Groups[gi], err = loadHoldingGroup(stmt, g, holdings)
```

Comment:
The usage of `g` here is really misleading. While it's technically correct, it would be much better to pass a pointer to the group data ( i.e. `&each.Groups[gi]` ) to loadHoldingGroup and let it perform the update.